src/provider/utils/async.ts

```diff
@@ -79,6 +79,48 @@ export function allowReject<T>(promise: Promise<T>): Promise<T> {
     return promise;
 }
 
+export async function raceUntilTrue<T>(items: T[], predicate: (v?: T) => boolean): Promise<T | null> {
+    let promisesCompleted = 0;
+    let errors = 0;
+    const valueFound = new DeferredPromise<T | null>();
+
+    const handleResolve = async (value?: T): Promise<T | void> => {
+        if (predicate(value)) {
+            valueFound.resolve(value);
+        }
+    };
+
+    const handleReject = (): void => {
+        errors++;
+    };
+
+    const completeCheck = async (value: T | void): Promise<T | null> => {
```

Comment:
Can this just return void, given the result of racePromises isn't used?

Reply:
Returns `Promise<void>`